examples/cpp/battleship/core.cpp

```diff
@@ -147,13 +147,13 @@ std::ostream& operator<<(std::ostream& os, const Position& pos) {
 
 std::ostream& operator<<(std::ostream& os, const GameState& state) {
   os << "GameState{";
-  for (int i = 0; i < NUM_SHIPS; i++) {
+  for (unsigned i = 0; i < NUM_SHIPS; i++) {
```

Comment:
Should i be unsigned, or should it be size_t?

Reply:
NUM_SHIPS is of type uint32_t and it looks like that's what's used elsewhere, so I'll update that.